crates/compiler/parse/src/expr.rs

```diff
@@ -623,8 +618,26 @@ pub fn parse_single_def<'a>(
             }
         }
         Err((MadeProgress, _)) => {
-            // a hacky way to get expression-based error messages. TODO fix this
-            Ok((NoProgress, None, initial))
+            // Try to parse as a Statement
+            match parse_statement_inside_def(
+                arena,
+                initial.clone(),
+                min_indent,
+                options,
+                start,
+                spaces_before_current_start,
+                // TODO including spaces_before_current here doubles things up
```

Comment:
I wasn't sure about the spaces parsing, AST and formatting, this seems to do the right thing but I haven't wrapped my head around it fully. The existing tests aren't impacted which gives me some assurance this is ok.